performance_profiling/train_base_nvtx.py

```diff
@@ -0,0 +1,284 @@
+from monai.inferers import sliding_window_inference
+from monai.losses import DiceLoss
+from monai.metrics import DiceMetric
+from monai.networks.layers import Norm
+from monai.networks.nets import UNet
+from monai.transforms import (
+    AsDiscrete,
+    Compose,
+    CropForegroundd,
+    EnsureChannelFirstd,
+    EnsureType,
+    EnsureTyped,
+    LoadImaged,
+    Orientationd,
+    RandCropByPosNegLabeld,
+    ScaleIntensityRanged,
+    Spacingd,
+)
+from monai.utils import Range, set_determinism
+
+nvidia_dlprof_pytorch_nvtx.init()
+
+root_dir = "./msd_data"
+out_dir = "./outputs_base"
+data_dir = os.path.join(root_dir, "Task09_Spleen")
+
+train_images = sorted(glob.glob(os.path.join(data_dir, "imagesTr", "*.nii.gz")))
+train_labels = sorted(glob.glob(os.path.join(data_dir, "labelsTr", "*.nii.gz")))
+data_dicts = [
+    {"image": image_name, "label": label_name}
+    for image_name, label_name in zip(train_images, train_labels)
+]
+train_files, val_files = data_dicts[:-9], data_dicts[-9:]
+
+set_determinism(seed=0)
+
+train_transforms = Compose(
+    [
+        Range("LoadImage")(LoadImaged(keys=["image", "label"])),
+        Range()(EnsureChannelFirstd(keys=["image", "label"])),
+        Range("Spacing")(
+            Spacingd(
+                keys=["image", "label"],
+                pixdim=(1.5, 1.5, 2.0),
+                mode=("bilinear", "nearest"),
+            )
+        ),
+        Range()(Orientationd(keys=["image", "label"], axcodes="RAS")),
+        Range()(
+            ScaleIntensityRanged(
+                keys=["image"],
+                a_min=-57,
+                a_max=164,
+                b_min=0.0,
+                b_max=1.0,
+                clip=True,
+            )
+        ),
+        Range()(CropForegroundd(keys=["image", "label"], source_key="image")),
+        Range("RandCrop")(
+            RandCropByPosNegLabeld(
+                keys=["image", "label"],
+                label_key="label",
+                spatial_size=(96, 96, 96),
+                pos=1,
+                neg=1,
+                num_samples=4,
+                image_key="image",
+                image_threshold=0,
+            )
+        ),
+        EnsureTyped(keys=["image", "label"]),
+    ]
+)
+
+val_transforms = Compose(
+    [
+        LoadImaged(keys=["image", "label"]),
+        EnsureChannelFirstd(keys=["image", "label"]),
+        Spacingd(
+            keys=["image", "label"],
+            pixdim=(1.5, 1.5, 2.0),
+            mode=("bilinear", "nearest"),
+        ),
+        Orientationd(keys=["image", "label"], axcodes="RAS"),
+        ScaleIntensityRanged(
+            keys=["image"],
+            a_min=-57,
+            a_max=164,
+            b_min=0.0,
+            b_max=1.0,
+            clip=True,
+        ),
+        CropForegroundd(keys=["image", "label"], source_key="image"),
+        EnsureTyped(keys=["image", "label"]),
+    ]
+)
+
+train_ds = CacheDataset(
```

Comment:
As @wyli mentioned, maybe you can use `DecathlonDataset` instead? It can automatically download and extract the spleen dataset.

Thanks.

Reply:
sure. will update it accordingly.